src/test/kotlin/com/atlassian/performance/tools/jiraactions/api/scenario/JiraRteEditPageTest.kt

```diff
@@ -0,0 +1,138 @@
+            clock = Clock.systemUTC(),
+            w3cPerformanceTimeline = DisabledW3cPerformanceTimeline()
+        )
+        val user = User("admin", "admin")
+        val userMemory = object : UserMemory {
+            override fun recall(): User {
+                return user
+            }
+
+            override fun remember(memories: Collection<User>) {
+                throw Exception("not implemented")
+            }
+        }
+
+        JiraCoreFormula.Builder()
+            .build()
+            .provision()
+            .use { jira ->
+                DockerisedChrome().start().use { browser ->
+                    val webJira = WebJira(
+                        browser.driver,
+                        jira.getUri(),
+                        user.password
+                    )
+                    val logInAction = scenario.getLogInAction(
+                        webJira,
+                        actionMeter,
+                        userMemory
+                    )
+                    val actions = scenario.getActions(
+                        webJira,
+                        SeededRandom(123),
+                        actionMeter
+                    )
+
+                    logInAction.run()
+                    actions.forEach { action ->
+                        action.run()
+                    }
+                }
+            }
+
+        val results = metrics.map { metric ->
+            metric.result
+        }
+        Assertions.assertThat(results).containsOnly(ActionResult.OK)
+    }
+}
+
+class JiraEditScenario constructor() : Scenario {
+
+    override fun getActions(jira: WebJira, seededRandom: SeededRandom, meter: ActionMeter): List<Action> {
+        val projectMemory = AdaptiveProjectMemory(random = seededRandom)
+        val jqlMemory = AdaptiveJqlMemory(seededRandom)
+        val issueKeyMemory = AdaptiveIssueKeyMemory(random = seededRandom)
+        val issueMemory = AdaptiveIssueMemory(issueKeyMemory, seededRandom)
+
+        val createIssue = CreateIssueAction(
+            jira = jira,
+            meter = meter,
+            seededRandom = seededRandom,
+            projectMemory = projectMemory
+        )
+        val searchWithJql = SearchJqlAction(
+            jira = jira,
+            meter = meter,
+            jqlMemory = jqlMemory,
+            issueKeyMemory = issueKeyMemory
+        )
+        val viewIssue = ViewIssueAction(
+            jira = jira,
+            meter = meter,
+            issueKeyMemory = issueKeyMemory,
+            issueMemory = issueMemory,
+            jqlMemory = jqlMemory
+        )
+        val editIssue = EditIssueAction(
+            jira = jira,
+            meter = meter,
+            issueMemory = issueMemory
+        )
+        val addComment = AddCommentAction(
+            jira = jira,
+            meter = meter,
+            issueMemory = issueMemory
+        )
+
+        val scenario: MutableList<Action> = mutableListOf(
```

Comment:
Do you use IntelliJ IDEA? If so, note that it will underline `scenario` in yellow, suggesting to  inline it. There's many other useful code hygiene hints, e.g. over the unnecessary `constructor()`.

Reply:
good to know :) thanks